src/main/rx/crsf.c

```diff
@@ -138,36 +167,45 @@ STATIC_UNIT_TESTED void crsfDataReceive(uint16_t c)
 
     if (crsfFramePosition < fullFrameLength) {
         crsfFrame.bytes[crsfFramePosition++] = (uint8_t)c;
-        crsfFrameDone = crsfFramePosition < fullFrameLength ? false : true;
-        if (crsfFrameDone) {
-            crsfFramePosition = 0;
+        if (crsfFramePosition == fullFrameLength) {
+            crsfFramePosition = 0;     
+            if (!crsfConsumerReady) {
+                resetRingBuffer();
+            } else {
+                rbufWriteData(crsfRingBuffer.ring, crsfFrame.bytes, fullFrameLength);
+                crsfRingBuffer.frames++;
+            }
         }
     }
 }
 
-STATIC_UNIT_TESTED uint8_t crsfFrameCRC(void)
+STATIC_UNIT_TESTED uint8_t crsfFrameStatus(void)
 {
-    // CRC includes type and payload
-    uint8_t crc = crc8_dvb_s2(0, crsfFrame.frame.type);
-    for (int ii = 0; ii < crsfFrame.frame.frameLength - CRSF_FRAME_LENGTH_TYPE_CRC; ++ii) {
-        crc = crc8_dvb_s2(crc, crsfFrame.frame.payload[ii]);
+    crsfFrame_t frame;
+
+    if (!crsfConsumerReady) {
+        crsfConsumerReady = true;
+        return RX_FRAME_PENDING;
     }
-    return crc;
-}
 
-STATIC_UNIT_TESTED uint8_t crsfFrameStatus(void)
-{
-    if (crsfFrameDone) {
-        crsfFrameDone = false;
-        if (crsfFrame.frame.type == CRSF_FRAMETYPE_RC_CHANNELS_PACKED) {
-            // CRC includes type and payload of each frame
-            const uint8_t crc = crsfFrameCRC();
-            if (crc != crsfFrame.frame.payload[CRSF_FRAME_RC_CHANNELS_PAYLOAD_SIZE]) {
-                return RX_FRAME_PENDING;
-            }
-            crsfFrame.frame.frameLength = CRSF_FRAME_RC_CHANNELS_PAYLOAD_SIZE + CRSF_FRAME_LENGTH_TYPE_CRC;
+    if (crsfRingBuffer.frames>0) {    
+        crsfRingBuffer.frames--;
+        rbufReadData(crsfRingBuffer.ring, &frame.bytes, 2);
+        const uint8_t length = frame.frame.frameLength;
+        if (length > 2 && length <= rbufReadBytesRemaining(crsfRingBuffer.ring)) {
+            rbufReadData(crsfRingBuffer.ring, &frame.frame.type, length);
+        } else {
+            crsfConsumerReady = false;
```

Comment:
Next call of `crsfFrameStatus` will set `crsfConsumerReady`, breaking ringbuffer synchronization (only 2 bytes were consumed, packet body remains)

Reply:
Can kill this... was an attempt to fix a nagging issue that did not work.